packages/imperative/src/config/__tests__/Config.api.unit.test.ts

```diff
@@ -264,6 +265,65 @@ describe("Config API tests", () => {
+                let caughtError: ImperativeError = {} as any;
+                try {
+                    config.api.layers.read();
+                } catch(err) {
+                    caughtError = err;
+                }
+                expect(existsSpy).toHaveBeenCalledTimes(5); // Once for each config layer and one more time for read
+                expect(readFileSpy).toHaveBeenCalledTimes(1);
+                expect(secureLoadSpy).toHaveBeenCalledTimes(0);
+                expect(jsoncParseSpy).toHaveBeenCalledTimes(1);
+                expect(caughtError.message).toContain("Please check this configuration file for errors");
+            });
+
+            it("should error when ignoreErrors is explicit false", async () => {
+                const config = await Config.load(MY_APP);
+                const existsSpy = jest.spyOn(fs, "existsSync").mockReturnValueOnce(true);
+                const readFileSpy = jest.spyOn(fs, "readFileSync");
+                const secureLoadSpy = jest.spyOn(config.api.secure, "loadCached");
+                const jsoncParseSpy = jest.spyOn(JSONC, "parse").mockImplementationOnce(() => { throw "failed to parse"; });
+                let caughtError: ImperativeError = {} as any;
+                try {
+                    config.api.layers.read({user: false, global: false, ignoreErrors: false});
+                } catch(err) {
+                    caughtError = err;
+                }
+                expect(existsSpy).toHaveBeenCalledTimes(5); // Once for each config layer and one more time for read
+                expect(readFileSpy).toHaveBeenCalledTimes(1);
+                expect(secureLoadSpy).toHaveBeenCalledTimes(0);
+                expect(jsoncParseSpy).toHaveBeenCalledTimes(1);
+                expect(caughtError.message).toContain("Please check this configuration file for errors");
+            });
+
+            it("should error when ignoreErrors is true", async () => {
+                const config = await Config.load(MY_APP);
+                const existsSpy = jest.spyOn(fs, "existsSync").mockReturnValueOnce(true);
+                const readFileSpy = jest.spyOn(fs, "readFileSync");
+                const secureLoadSpy = jest.spyOn(config.api.secure, "loadCached");
+                const jsoncParseSpy = jest.spyOn(JSONC, "parse").mockImplementationOnce(() => { throw "failed to parse"; });
+                const redSpy = jest.fn().mockImplementation((str: string) => str);
+                jest.spyOn(TextUtils, "chalk", "get").mockReturnValueOnce({red: redSpy});
+                let caughtError;
+                try {
+                    config.api.layers.read({user: false, global: false, ignoreErrors: true});
+                } catch(err) {
+                    caughtError = err;
+                }
+                expect(existsSpy).toHaveBeenCalledTimes(5); // Once for each config layer and one more time for read
+                expect(readFileSpy).toHaveBeenCalledTimes(1);
```

Comment:
@t1m0thyj @zFernand0 im not sure why this case requires 6 calls instead of 5 like the rest. making sure this is OK with you guys

Reply:
I think the extra call to `existsSync` may be happening because the logger isn't actually being mocked. See my other comment for why.